### PR TITLE
[TACHYON-1610] User thread safety annotation in the common module

### PR DESCRIPTION
**Pull request by jsimsa**
Addressed: https://tachyon.atlassian.net/browse/TACHYON-1610

### CODE REVIEW

File: common/src/main/java/tachyon/conf/TachyonConf.java [hunk 1]
```diff
@@ -68,6 +70,7 @@
  * The class only supports creation using {@code new TachyonConf(properties)} to override default
  * values.
  */
+@ThreadSafe
```
Comment (apc999) [inline common/src/main/java/tachyon/conf/TachyonConf.java]: This class is not threadsafe. See setters for `mProperties`

File: common/src/main/java/tachyon/underfs/UnderFileSystemRegistry.java [hunk 2]
```diff
@@ -71,6 +73,7 @@
  * will have the first opportunity to indicate whether they support a requested path.
  * </p>
  */
+@ThreadSafe
```
Comment (apc999) [inline common/src/main/java/tachyon/underfs/UnderFileSystemRegistry.java]: this class is not thread safe, 
see potential race between `register` and `unregister`

File: common/src/main/java/tachyon/util/io/ByteIOUtils.java [hunk 3]
```diff
@@ -19,9 +19,12 @@
 import java.io.OutputStream;
 import java.nio.ByteBuffer;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * A collection of utility functions to read/write primitive values from/to a byte array.
  */
+@NotThreadSafe
```
Comment (apc999) [inline common/src/main/java/tachyon/util/io/ByteIOUtils.java]: this class should be thread-safe

**Comment by apc999** [comment]
thanks for the improvement, I left a few comments!

File: common/src/main/java/tachyon/conf/TachyonConf.java (hunk 1 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/conf/TachyonConf.java]: done

**Comment by apc999** [comment]
+1 LGTM

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**